craft_application/commands/lifecycle.py

```diff
@@ -86,15 +87,23 @@ def fill_parser(self, parser: argparse.ArgumentParser) -> None:
         super().fill_parser(parser)
 
         group = parser.add_mutually_exclusive_group()
-        group.add_argument(
-            "--shell",
-            action="store_true",
-            help="Shell into the environment in lieu of the step to run.",
-        )
-        group.add_argument(
-            "--shell-after",
+
+        if self._should_add_shell_args():
```

Comment:
Is there a situation where we don't want shell options to be available?

Reply:
The applications today don't use those for "pack"

Reply:
Ah, that makes sense. One can argue that `pack --shell` is the equivalent to `prime --shell-after`, but `pack --shell-after` wouldn't make much sense anyway.